File: influx-service/src/test/java/renovation/influx/service/controller/StartupControllerTest.java
/*
 * Created under not commercial project "Renovation"
 *
 * Copyright 2021-2025
 */

package renovation.influx.service.controller;

import org.apache.http.HttpStatus;
import org.hamcrest.CoreMatchers;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import renovation.common.web.RestTestInit;
import renovation.influx.service.web.Route;

import static renovation.common.util.JsonUtil.simplify;

@Tag("componentTest")
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class StartupControllerTest extends RestTestInit {

    public StartupControllerTest() {
        super(Route.STARTUP);
    }

    @Test
    void about() {
        getRequest().get("/about")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(CoreMatchers.equalTo(
                        simplify(
                                """
                                        {
                                            "name"        : "influx-service",
                                            "description" : "Influx service module"
                                        }
                                        """)
                ));
    }

    @Test
    void index() {
        getRequest().get("/module")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(CoreMatchers.equalTo("Hi, it's \"Influx-service\" module"));
    }
}

File: backend/src/test/kotlin/renovation/backend/web/StartupControllerTest.kt
/*
 * Created under not commercial project "Renovation"
 *
 * Copyright 2021-2024
 */

package renovation.backend.web

import io.restassured.RestAssured
import io.restassured.module.kotlin.extensions.Then
import io.restassured.module.kotlin.extensions.When
import kotlin.test.BeforeTest
import kotlin.test.Test
import org.apache.http.HttpStatus.SC_OK
import org.hamcrest.CoreMatchers
import org.junit.jupiter.api.Tag
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.server.LocalServerPort
import org.springframework.test.context.ContextConfiguration
import renovation.backend.KeycloakContainerConfig
import renovation.backend.PostgresContainerConfig
import renovation.backend.RedisContainerConfig

@Tag("integrationTest")
@ContextConfiguration(
    classes = [
        KeycloakContainerConfig::class,
        PostgresContainerConfig::class,
        RedisContainerConfig::class
    ]
)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
internal class StartupControllerTest(
    @LocalServerPort val port: Int
) {
    @BeforeTest
    fun init() {
        RestAssured.port = port
    }

    @Test
    fun `Assert startup controller`() {
        When {
            get("/project")
        }.Then {
            statusCode(SC_OK)
            body(CoreMatchers.equalTo("Hi, it's \"Renovation backend\" project"))
        }
    }

    @Test
    fun `Assert about controller`() {
        When {
            get("/about")
        }.Then {
            statusCode(SC_OK)
            body(
                CoreMatchers.equalTo(
                    """{"name":"renovation backend module",""" +
                        """"description":"Main backend part of renovation project"}"""
                )
            )
        }
    }
}
